File: elixir/easy/0001/0001.exs
defmodule Solution do
  @spec two_sum(nums :: [integer], target :: integer) :: [integer]
  def two_sum(nums, target) do
    nums_indexed = Enum.with_index(nums)

    result =
      for x <- nums_indexed,
          y <- nums_indexed,
          elem(x, 1) < elem(y, 1) and elem(x, 0) + elem(y, 0) == target,
          do: [elem(x, 1), elem(y, 1)]

    [head | tail] = result
    head
  end
end

ExUnit.start()

defmodule AssertionTest do
  use ExUnit.Case, async: true

  test "test_1" do
    assert Solution.two_sum([2, 7, 11, 15], 9) == [0, 1]
  end

  test "test_2" do
    assert Solution.two_sum([3, 2, 4], 6) == [1, 2]
  end

  test "test_3" do
    assert Solution.two_sum([3, 3], 6) == [0, 1]
  end
end

File: py/easy/0001/0001.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        len_nums = len(nums)
        for i in range(len_nums):
            for j in range(i + 1, len_nums):
                if nums[i] + nums[j] == target:
                    return [i, j]


solution = Solution()

assert solution.twoSum(nums=[2, 7, 11, 15], target=9) == [0, 1]
assert solution.twoSum(nums=[3, 2, 4], target=6) == [1, 2]
assert solution.twoSum(nums=[3, 3], target=6) == [0, 1]
